frontend/src/metabase/reference/components/FieldTypeDetail.jsx

```diff
@@ -25,25 +23,24 @@ const FieldTypeDetail = ({
       <div className={D.detailTitle}>
         <span>{t`Field type`}</span>
       </div>
-      <div className={cx(D.detailSubtitle, { [CS.mt1]: true })}>
+      <div className={D.detailSubtitle}>
         <span>
           {isEditing ? (
-            <Select
-              placeholder={t`Select a field type`}
-              value={fieldTypeFormField.value || field.semantic_type}
-              options={FIELD_SEMANTIC_TYPES.concat({
-                id: null,
-                name: t`No field type`,
-                section: t`Other`,
-              }).filter(type =>
-                !isNumericBaseType(field)
-                  ? !(type.id && type.id.startsWith("timestamp_"))
-                  : true,
-              )}
-              optionValueFn={o => o.id}
-              onChange={({ target: { value } }) =>
-                fieldTypeFormField.onChange(value)
+            <SemanticTypePicker
+              field={field}
+              value={
+                typeof fieldTypeFormField.value !== "undefined"
+                  ? fieldTypeFormField.value
+                  : field.semantic_type
```

Comment:
`fieldTypeFormField.value || field.semantic_type` would make it impossible to select "No semantic type" option (which has `null` value) - this fixes it.